numpyro/distributions/transforms.py

```diff
@@ -1044,14 +1044,14 @@ class ConstraintRegistry(object):
     def __init__(self):
         self._registry = {}
 
-    def register(self, constraint, factory=None):
+    def register(self, constraint_list, factory=None):
         if factory is None:
-            return lambda factory: self.register(constraint, factory)
+            return lambda factory: self.register(constraint_list, factory)
 
-        if isinstance(constraint, constraints.Constraint):
-            constraint = type(constraint)
-
-        self._registry[constraint] = factory
+        for constraint in constraint_list:
+            if isinstance(constraint, constraints.Constraint):
+                constraint = type(constraint)
+            self._registry[constraint] = factory
```

Comment:
Oops, really sorry, I think we need to return `factory` here. Could you revert to the previous commit and add
```
def register(...):
    ...
    return factory
```

Reply:
Ah, right - that works, thanks!